core/src/builtin_widgets/anchor.rs

```diff
@@ -1,92 +1,183 @@
-  #[declare(builtin, default)]
-  pub bottom_anchor: f32,
+  /// positions the widget's bottom edge x pixels above the target's bottom
+  /// edge.
+  Bottom(f32),
 }
 
-impl Render for LeftAnchor {
-  fn perform_layout(&self, clamp: BoxClamp, ctx: &mut LayoutCtx) -> Size {
-    let mut layouter = ctx.assert_single_child_layouter();
-    let child_size = layouter.perform_widget_layout(clamp);
-    let left = self.left_anchor;
-    layouter.update_position(Point::new(left, 0.));
-    Size::new((child_size.width + left).max(0.), child_size.height)
+impl HAnchor {
+  pub fn map(self, f: impl FnOnce(f32) -> f32) -> Self {
+    match self {
+      HAnchor::Left(x) => HAnchor::Left(f(x)),
+      HAnchor::Right(x) => HAnchor::Right(f(x)),
+    }
   }
+}
 
-  fn paint(&self, _: &mut PaintingCtx) {}
-
-  fn hit_test(&self, _: &HitTestCtx, _: Point) -> HitTest {
-    HitTest { hit: false, can_hit_child: true }
+impl VAnchor {
+  pub fn map(self, f: impl FnOnce(f32) -> f32) -> Self {
+    match self {
+      VAnchor::Top(x) => VAnchor::Top(f(x)),
+      VAnchor::Bottom(x) => VAnchor::Bottom(f(x)),
+    }
   }
 }
 
-impl Render for RightAnchor {
-  fn perform_layout(&self, clamp: BoxClamp, ctx: &mut LayoutCtx) -> Size {
-    let mut layouter = ctx.assert_single_child_layouter();
-    let child_size = layouter.perform_widget_layout(clamp);
-    let right = self.right_anchor;
-    let x = clamp.max.width - child_size.width - right;
-    layouter.update_position(Point::new(x, 0.));
+impl PartialEq for HAnchor {
+  fn eq(&self, other: &Self) -> bool {
+    match (self, other) {
+      (HAnchor::Left(x1), HAnchor::Left(x2)) => (x1 - x2).abs() < f32::EPSILON,
+      (HAnchor::Right(x1), HAnchor::Right(x2)) => (x1 - x2).abs() < f32::EPSILON,
+      _ => false,
+    }
+  }
+}
 
-    Size::new((child_size.width + x).max(0.), child_size.height)
+impl PartialEq for VAnchor {
+  fn eq(&self, other: &Self) -> bool {
+    match (self, other) {
+      (VAnchor::Top(y1), VAnchor::Top(y2)) => (y1 - y2).abs() < f32::EPSILON,
+      (VAnchor::Bottom(y1), VAnchor::Bottom(y2)) => (y1 - y2).abs() < f32::EPSILON,
+      _ => false,
+    }
   }
+}
 
-  fn paint(&self, _: &mut PaintingCtx) {}
+#[derive(Clone, Copy, Default)]
+pub struct Anchor {
+  /// Specifies the horizontal position you want to anchor the widget, See
+  /// [`HAnchor`]!. if None, the widget is anchored by the parent
+  pub x: Option<HAnchor>,
 
-  fn hit_test(&self, _: &HitTestCtx, _: Point) -> HitTest {
-    HitTest { hit: false, can_hit_child: true }
+  /// Specifies the vertical position you want to anchor the widget, See
+  /// [`VAnchor`]! if None, the widget is anchored by the parent
+  pub y: Option<VAnchor>,
+}
+
+impl Lerp for HAnchor {
+  fn lerp(&self, other: &Self, t: f32) -> Self {
+    match (self, other) {
+      (HAnchor::Left(x1), HAnchor::Left(x2)) => HAnchor::Left(x1.lerp(x2, t)),
+      (HAnchor::Right(x1), HAnchor::Right(x2)) => HAnchor::Right(x1.lerp(x2, t)),
+      _ => unreachable!(),
```

Comment:
Right None, None Left  should implement  `Lerp`, and should not panic when hit left <-->right